packages/mobx/src/types/actionannotation.ts

```diff
@@ -58,6 +65,80 @@ function extend_(
     return adm.defineProperty_(key, actionDescriptor, proxyTrap)
 }
 
+function decorate_20223_(this: Annotation, mthd, context: DecoratorContext) {
+    assert20223DecoratorType(context, ["method", "accessor"])
+    const { kind, name, addInitializer } = context
+    const ann = this
+
+    const _createAction = m =>
+        createAction(ann.options_?.name ?? name!.toString(), m, ann.options_?.autoAction ?? false)
+
+    // Backwards/legacy behavior
+    if (kind == "field") {
+        addInitializer(function () {
+            const adm: ObservableObjectAdministration = asObservableObject(this)[$mobx]
+            const self = this as any
+            const actionDescriptor = createActionDescriptor(
+                adm,
+                self,
+                name,
+                { value: self[name] },
+                false
+            )
+            defineProperty(self, name, actionDescriptor)
+        })
+        return
+    }
+
+    if (kind == "method") {
+        addInitializer(function () {
+            const adm: ObservableObjectAdministration = asObservableObject(this)[$mobx]
+            const target = adm.proxy_ ?? adm.target_
+            const self = this as any
+            let actionMethod = self[name]
+            if (isAction(actionMethod)) return
+
+            if (ann.options_?.bound) {
+                actionMethod = actionMethod.bind(target)
+            }
+
+            self[name] = _createAction(actionMethod)
+
+            Object.defineProperty(self, name, {
+                enumerable: false
+            })
+        })
+        return
```

Comment:
I'll take a loser look Monday. I learned that `addInitializer` doesn't behave the way that I'd immediately expect it to (all initializers run before fields have been initialized instead of being interleaved), so it changes what's made available when and is causing some complications. My answer to those complications will likely affect the answer to this.